nixnet/database/_database_object_base.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import typing  # NOQA: F401
+
+
+class DatabaseObjectBase(object):
```

Comment:
Since this should be public, we should ensure we have a name we like.  I personally find -`Base` suffixes a little off, at least in python code.

Reply:
I was following what I saw with [SessionBase](https://github.com/ni/nixnet-python/blob/master/nixnet/_session/base.py#L22). So your suggestion is `DatabaseObject`?

Reply:
iirc SessionBase was me doing what I said not to do, concrete inheritance for code sharing.  Its not actually exposed in the API.

Something like `DatabaseObject` might work.